### PR TITLE
Feat/datefield

### PR DESCRIPTION
**Pull request by danielzairin**
RadixUI doesn't have a primitive Date component so I used [react-aria-components' DateField](https://react-spectrum.adobe.com/react-aria/DateField.html). If this is an issue I can look at other solutions.

react-aria uses the `@internationalized/date` package for date values. Since our DatePicker component uses native Date objects, I've adapted the props for consistency. To keep things simple, I've also remove validation props like "minValue", "maxValue" since these expect date instances from the `@internationalized/date` package too.

https://github.com/user-attachments/assets/0520a897-ad01-44cf-b179-529eae558f1d

### CODE REVIEW

**Comment by danielzairin** [comment]
Update: Used html date input as discussed with @irfancoder.

Got a few issues that I think are minor:
 - I'm unable to style each datefield segment's text to be gray when there's no value, tried various CSS selectors to get the "no value" state but none worked
 - On Safari, the text color is always gray when the input is disabled. I can't seem to override this, even with `!important`

File: pnpm-lock.yaml
Comment (danielzairin) [inline pnpm-lock.yaml:1]: Not sure why pnpm re-created the pnpm-lock file when I run `pnpm i`. I'm using pnpm 8.15.6.

Comment (irfancoder) [reply pnpm-lock.yaml:1]: time to upgrade buddy 😆 (at 9.7.0)

**Review by irfancoder — APPROVED** [review]
This is good enough. tyvm. Will let you know if I have an idea how to solve the minor issues

**Pull request merged**